server/src/apub/mod.rs

```
@@ -139,24 +151,70 @@ pub trait FromApub {
     Self: Sized;
 }
 
+pub trait ApubObjectType {
+  fn send_create(&self, creator: &User_, conn: &PgConnection) -> Result<(), Error>;
+  fn send_update(&self, creator: &User_, conn: &PgConnection) -> Result<(), Error>;
+}
+
+pub fn get_shared_inbox(actor_id: &str) -> String {
+  let url = Url::parse(actor_id).unwrap();
+  format!(
+    "{}://{}{}/inbox",
+    &url.scheme(),
+    &url.host_str().unwrap(),
+    if let Some(port) = url.port() {
+      format!(":{}", port)
+    } else {
+      "".to_string()
+    },
+  )
+}
+
 pub trait ActorType {
   fn actor_id(&self) -> String;
 
   fn public_key(&self) -> String;
 
+  // These two have default impls, since currently a community can't follow anything,
+  // and a user can't be followed (yet)
```

Better throw an error then, or use something like `unimplemented!()`. Failing silently sucks.